source/code/source/playfab/PlayFabPluginManager.cpp

```diff
@@ -27,8 +27,8 @@ namespace PlayFab
 
     std::shared_ptr<IPlayFabPlugin> PlayFabPluginManager::GetPluginInternal(const PlayFabPluginContract contract, const std::string& instanceName)
     {
-        const auto key = std::make_pair(contract, instanceName);
-        auto pluginEntry = plugins.find(key);
+        std::pair<PlayFabPluginContract, std::string> key = std::make_pair(contract, instanceName);
+        const auto& pluginEntry = plugins.find(key);
```

Comment:
>const auto& [](start = 8, length = 11)

similar comment as elsewhere. find doesn't return an reference type. it returns a value type so this may be misleading. My guess is you used auto because the full iterator type was long and ugly. I empathize but don't think using `const auto&` here is better than `auto` or using the full type

Reply:
the rules in your description say "auto" is okay for iterators. I don't mind that but similar to other comments it's worth reconsidering the const ref portion of the type since it doesn't represent the return value of the function being called.

---
In reply to: [340922982](https://github.com/PlayFab/XPlatCoreTemplate/pull/33#discussion_r340922982) [](ancestors = 340922982)